operations/helm/charts/mimir-distributed/templates/_helpers.tpl

```diff
@@ -429,6 +432,38 @@ Examples:
 {{- $section | toYaml -}}
 {{- end -}}
 
+{{/*
+Return the Vault Agent pod annotations if enabled and required by the component
+mimir.vaultAgent.annotations takes 2 arguments
+  .ctx = the root context of the chart
+  .component = the name of the component
+*/}}
+{{- define "mimir.vaultAgent.annotations" -}}
+{{- $vaultEnabledComponents := dict
```

Comment:
Is there any component that isn't tls enabled? If there isn't,we can drop this dict and just use `vaultAgent.enabled` to globally enable or disable annotation to all mimir components.